src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -172,6 +173,10 @@ def analyze(
             so that it will detect column name change.
             however, in the result calculation, we want to column name to be the output name, which is 'totB',
             so we set keep_alias to False in the execution.
+            quote_literal: if true, the literal string will be returned single quoted as '<literal>', column name
+            can be like e.g., <col> like '<literal>', <col> REGEXP '<literal>'.
+            quote_literal is set to True when `analyze` is called to describe the column name of a dataframe rather
+            than calculation, this is to align with the behavior of live connection describing a dataframe.
```

Comment:
I found this a bit confusing, I thought `analyze` is always called to describe the resulting column name of an expression. `calculate_expression` is what actually computes.

Reply:
yeah your understanding is correct.
I think this parameter needs a better name, probably something like  `quote_literal_in_column_name`.

so when we get the column name, single quote is added around the literal in the column name, e.g., in our test code "V%" is single quoted:
```python
def test_like(session):
    df1 = session.create_dataframe(["v"], schema=["c"])
    df2 = df1.select(df1["c"].like(lit("v%")))

    assert (
        df2._output[0].name
        == df2.columns[0]
        == get_metadata_names(session, df2)[0]
        == '"""C"" LIKE \'V%\'"'
    )
```

but for calculation, we are just comparing the string "%V", no single quotes.

Reply:
updated to use `calculate_expression` to literal computation and single quote literal all the time in analyze